src/core/transaction.zig

```diff
@@ -393,47 +393,12 @@ pub const Message = struct {
         return index < self.signature_count;
     }
 
-    const RESERVED_ACCOUNTS: []const Pubkey = &.{
-        // builtin programs
-        sig.runtime.program.bpf_loader.v2.ID,
-        sig.runtime.program.bpf_loader.v1.ID,
-        sig.runtime.program.bpf_loader.v3.ID,
-        sig.runtime.program.config.ID,
-
-        sig.runtime.ids.FEATURE_PROGRAM_ID,
-        sig.runtime.ids.CONFIG_PROGRAM_STAKE_CONFIG_ID,
-        sig.runtime.program.stake.ID,
-        sig.runtime.program.system.ID,
-        sig.runtime.program.vote.ID,
-        sig.runtime.program.zk_elgamal.ID,
-        sig.runtime.ids.ZK_TOKEN_PROOF_PROGRAM_ID,
-
-        // sysvars
-        sig.runtime.sysvar.Clock.ID,
-        sig.runtime.sysvar.EpochSchedule.ID,
-        sig.runtime.sysvar.Fees.ID,
-        sig.runtime.ids.SYSVAR_INSTRUCTIONS_ID,
-        sig.runtime.sysvar.RecentBlockhashes.ID,
-        sig.runtime.sysvar.Rent.ID,
-        sig.runtime.ids.SYSVAR_REWARDS_ID,
-        sig.runtime.sysvar.SlotHashes.ID,
-        sig.runtime.sysvar.SlotHistory.ID,
-        sig.runtime.sysvar.StakeHistory.ID,
-
-        // other
-        sig.runtime.ids.NATIVE_LOADER_ID,
-    };
-
     pub fn isWritable(self: Message, index: usize) bool {
         const is_readonly_signed =
             index < self.signature_count and
             index >= self.signature_count - self.readonly_signed_count;
         const is_readonly_unsigned = index >= self.account_keys.len - self.readonly_unsigned_count;
-        const is_writable = !(is_readonly_signed or is_readonly_unsigned);
-
-        return is_writable and for (RESERVED_ACCOUNTS) |reserved| {
-            if (reserved.equals(&self.account_keys[index])) break false;
-        } else true;
+        return !(is_readonly_signed or is_readonly_unsigned);
     }
```

Comment:
Why did you remove this? This seems like the best place for this code since it's the most widely applicable function that's likely to be used in the most places for determining writability. Also this is where agave has this logic.

Reply:
Its equivalent to agave's [`Message.is_writable_index`](https://github.com/anza-xyz/solana-sdk/blob/5ff67c1a53c10e16689e377f98a92ba3afd6bb7c/message/src/versions/v0/loaded.rs#L118), but its `is_writable()` actually hits a hashset thats built with that + [`is_writable_internal`](https://github.com/anza-xyz/solana-sdk/blob/5ff67c1a53c10e16689e377f98a92ba3afd6bb7c/message/src/versions/v0/loaded.rs#L139-L150) which is what this code adds. But doing that requires the ALUT accounts in addition to `Message.account_keys` (for computing `is_upgradeable_loader_present`). Those aren't available in our `Message` type, so opted to instead do it where they are + only place where `isWritable` is called: `resolveTransaction()`

Reply:
I see. It's tricky to come up with a clean general solution here.

Would you mind renaming `isWritable` to `isWritableIndex` and adding a doc comment describing its limited role?

Reply:
https://github.com/Syndica/sig/pull/836/commits/fa2d36d9579a4275bf5c328dc488504df518d1d0

Reply:
nice